main.go

```diff
@@ -47,14 +55,32 @@ func init() {
 }
 
 func main() {
+	// metricsAddr is used by Prometheus to gather the NFD Operator resource usage data.
+	// The bind address tells Prometheus which port to scrape this data's metrics from.
 	var metricsAddr string
+
+        // enableLeaderElection should be set to 'disable' by default If we enable leader
+        // election, then only one node can run the controller manager and we will not
+        // have NFD Operator running on all nodes.
 	var enableLeaderElection bool
+
+        // probeAddr is responsible for the health probe bind address, where the health
+        // probe is responsible for determining liveness, readiness, and configuration 
+        // of the operator pods.
```

Comment:
I think these comments are kind of unneeded as they are explained in the `flag.*Var()` below. If we want to improve the documentation a better place might be to directly edit the "description" arg of the flag calls.

Reply:
Thanks. I've moved this information to the flag descriptions and made it more concise. Let me know what you think.